pkg/controller/podautoscaler/horizontal.go

```diff
@@ -63,92 +70,101 @@ func (a *HorizontalController) Run(syncPeriod time.Duration) {
 	}, syncPeriod, util.NeverStop)
 }
 
-func (a *HorizontalController) reconcileAutoscalers() error {
-	ns := api.NamespaceAll
-	list, err := a.client.Experimental().HorizontalPodAutoscalers(ns).List(labels.Everything(), fields.Everything())
+func (a *HorizontalController) reconcileAutoscaler(hpa experimental.HorizontalPodAutoscaler) error {
+	reference := fmt.Sprintf("%s/%s/%s", hpa.Spec.ScaleRef.Kind, hpa.Spec.ScaleRef.Namespace, hpa.Spec.ScaleRef.Name)
+
+	scale, err := a.client.Experimental().Scales(hpa.Spec.ScaleRef.Namespace).Get(hpa.Spec.ScaleRef.Kind, hpa.Spec.ScaleRef.Name)
 	if err != nil {
-		return fmt.Errorf("error listing nodes: %v", err)
+		a.eventRecorder.Event(&hpa, "failed to query scale subresource", err.Error())
```

Comment:
Fixed.

I was basing on events generated by service controller. But after careful investigation, our convention is different: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#events, and events generated by service controller are not correct. Thanks for noticing this!